### ISSUE TITLE
Prompt request when using github.copilot.chat.experimental.codeGeneration.instructions 

### ISSUE THREAD

**Issue by brianpajak-intel** [microsoft/vscode-copilot-release#1689]
I love that the development team implemented [github.copilot.chat.experimental.codeGeneration.instructions](vscode://settings/github.copilot.chat.experimental.codeGeneration.instructions).  We do lots of coding using a proprietary language similar to TCL and Copilot struggles with some of the constructs.  The language uses different procedures for common commands (e.x. display_message instead of puts).  I have been using the new feature to steer Copilot to swap in custom commands when it generates code.  It does ok.  I would say it is still hit or miss, but since this is a new feature I am sure it will improve over time.  What I would like to see is the ability to guide the tool dynamically.  In our language we have the concept of collections.  When iterating over elements in a collection you use the command foreach_in_collection instead of the standard foreach command.  I would love it if I could guide Copilot to use foreach whenever it knows the variable being acted on is a list, or to use foreach_in_collection whenever the variable being acted on is not a list, **OR** have it ask me whether I want to use a foreach or foreach_in_collection when it is generating the code / creating ghost text.  This way I don't need to go back and modify things.  Something like this:

(1. foreach or 2. foreach_in_collection) element $my_collection {

Then I could select 1 or 2 and have it just add the correct one.  Something to save me from having to back up and edit. 

Brian

**Comment by aeschli** [comment]
You want Copilot to ask you back how to fix the issue?

**Comment by brianpajak-intel** [comment]
Yes.  Exactly. I am using code generation instructions and Copilot still uses for and not foreach_in_collection even though I have tried to guide it:

- Whenever operating on a variable which is a collection use foreach_in_collection instead of foreach when iterating over elements.

  Example of when to use foreach

  ```
  set my_list [list a b c]
  foreach list_item $list {
  }
  ```

  Example of when to use foreach_in_collection

  ```
  set my_collection [get_ports *]
  foreach_in_collection port $ports {
  }
  ```

It would be able to provide some guidance to copilot when it may not know for sure or if I am guiding it incorrectly in my copilot-instructions.md please let me know.  We have lots of proprietary code and Copilot struggles with some of the syntax.  

But, I do want to say that even though it struggles I am still SUPER happy with how well Copilot does work and give a huge thank you to the dev team for helping me reduce my work with this incredible tool.  Thank you!  Copilot is fantastic.
